pyfixest/did/did2s.py

```diff
@@ -126,37 +128,34 @@ def _did2s_estimate(
         treatment (str): The name of the treatment variable. Must be boolean.
         i_ref1 (int, str or list): The reference value(s) for the first variable used with "i()" syntax. Only applicable for the second stage formula.
         i_ref2 (int, str or list): The reference value(s) for the second variable used with "i()" syntax. Only applicable for the second stage formula.
-    Returns:
+
+    Returns
+    -------
         A fitted model object of class feols and the first and second stage residuals.
     """
-
     _first_stage_full = f"{yname} {_first_stage}"
     _second_stage_full = f"{yname}_hat {_second_stage}"
 
     if treatment is not None:
         if treatment not in data.columns:
             raise ValueError(f"The variable {treatment} is not in the data.")
         # check that treatment is boolean
-        if data[treatment].dtype != "bool":
-            if data[treatment].dtype in [
-                "int64",
-                "int32",
-                "int8",
-                "float64",
-                "float32",
-            ]:
-                if data[treatment].nunique() == 2:
-                    data[treatment] = data[treatment].astype(bool)
-                    warnings.warn(
-                        f"The treatment variable {treatment} was converted to boolean."
-                    )
-                else:
-                    raise ValueError(
-                        f"The treatment variable {treatment} must be boolean."
-                    )
-        _not_yet_treated_data = data[data[treatment] == False]
+        if data[treatment].dtype != "bool" and data[treatment].dtype in [
+            "int64",
+            "int32",
+            "int8",
+            "float64",
+            "float32",
+        ]:
+            if data[treatment].nunique() != 2:
+                raise ValueError(f"The treatment variable {treatment} must be boolean.")
+            data[treatment] = data[treatment].astype(bool)
+            warnings.warn(
+                f"The treatment variable {treatment} was converted to boolean."
+            )
+        _not_yet_treated_data = data[data[treatment] is False]
     else:
-        _not_yet_treated_data = data[data["ATT"] == False]
+        _not_yet_treated_data = data[data["ATT"] is False]
```

Comment:
@NKeleher this is where the KeyError for the `did2s` test stem from.

Reply:
In fact I believe it's the same error in all of these: 

FAILED tests/test_did.py::test_event_study - KeyError: False -> also tests the did2s implementation
FAILED tests/test_did.py::test_did2s - KeyError: False -> error in did2s
FAILED tests/test_did.py::test_errors - KeyError: False -> error in error checking for did2s